pkg/volume/quobyte/quobyte.go

```diff
@@ -0,0 +1,247 @@
+	return &quobyteUnmounter{&quobyte{
+		volName: volName,
+		mounter: mounter,
+		pod:     &api.Pod{ObjectMeta: api.ObjectMeta{UID: podUID}},
+		plugin:  plugin,
+	}}, nil
+}
+
+// Quobyte volumes represent a bare host directory mount of an quobyte export.
+type quobyte struct {
+	volName string
+	pod     *api.Pod
+	user    string
+	group   string
+	mounter mount.Interface
+	plugin  *quobytePlugin
+	volume.MetricsNil
+}
+
+type quobyteMounter struct {
+	*quobyte
+	registry string
+	volume   string
+	readOnly bool
+}
+
+var _ volume.Mounter = &quobyteMounter{}
+
+func (mounter *quobyteMounter) GetAttributes() volume.Attributes {
+	return volume.Attributes{
+		ReadOnly:        mounter.readOnly,
+		Managed:         false,
+		SupportsSELinux: false,
+	}
+}
+
+// SetUp attaches the disk and bind mounts to the volume path.
+func (mounter *quobyteMounter) SetUp(fsGroup *int64) error {
+	pluginDir := mounter.plugin.host.GetPluginDir(strings.EscapeQualifiedNameForDisk(quobytePluginName))
+	return mounter.SetUpAt(pluginDir, fsGroup)
+}
+
+func (mounter *quobyteMounter) SetUpAt(dir string, fsGroup *int64) error {
+	// Check if Quobyte is already mounted on the host in the Plugin Dir
+	// if so we can use this mountpoint instead of creating a new one
+	// IsLikelyNotMountPoint wouldn't check the mount type
+	if mounted, err := mounter.pluginDirIsMounted(dir); err != nil {
+		return err
+	} else if mounted {
+		return nil
+	}
+
+	os.MkdirAll(dir, 0750)
+	var options []string
+	if mounter.readOnly {
+		options = append(options, "ro")
+	}
+
+	if err := mounter.mounter.Mount(path.Join(mounter.registry, mounter.volume), dir, "quobyte", options); err != nil {
+		return fmt.Errorf("quobyte: mount failed: %v", err)
+	}
+
+	glog.V(4).Infof("quobyte: mount set up: %s", dir)
+
+	return nil
+}
+
+// GetPath returns the path to the user specific mount of a Quobyte volume
+// Returns a path in the format ../user@volume e.g. ../root@MyVolume
+// or if a group is set ../user#group@volume
+func (quobyteVolume *quobyte) GetPath() string {
+	user := quobyteVolume.user
+	if len(user) == 0 {
+		user = "root"
+	}
+
+	// Quobyte has only one mount in the PluginDir where all Volumes are mounted
+	// The Quobyte client does a fixed-user mapping
+	pluginDir := quobyteVolume.plugin.host.GetPluginDir(strings.EscapeQualifiedNameForDisk(quobytePluginName))
+	if len(quobyteVolume.group) > 0 {
+		return path.Join(pluginDir, fmt.Sprintf("%s#%s@%s", user, quobyteVolume.group, quobyteVolume.volName))
+	}
+
+	return path.Join(pluginDir, fmt.Sprintf("%s@%s", user, quobyteVolume.volName))
+}
+
+type quobyteUnmounter struct {
+	*quobyte
+}
+
+var _ volume.Unmounter = &quobyteUnmounter{}
+
+func (unmounter *quobyteUnmounter) TearDown() error {
+	return unmounter.TearDownAt(unmounter.GetPath())
+}
+
+// We don't need to unmount on the host because only one mount exists
+func (unmounter *quobyteUnmounter) TearDownAt(dir string) error {
+	return nil
```

Comment:
We will never unmount all Quobyte Volumes from the host because this would require some kind of host specific counter which would be more complex to handle than just leave one mount point on the host.